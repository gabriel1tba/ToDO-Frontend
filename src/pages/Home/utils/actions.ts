import { TTodo } from 'services/TodoService/interfaces';

export enum Types {
  GET_TODOS = 'GET_TODOS',
  CREATE_TODO = 'CREATE_TODO',
  UPDATE_TODO = 'UPDATE_TODO',
  DELETE_TODO = 'DELETE_TODO',
}

interface GetTodos {
  type: Types.GET_TODOS;
  payload: TTodo[];
}

interface CreateTodo {
  type: Types.CREATE_TODO;
  payload: TTodo;
}

interface UpdateTodo {
  type: Types.UPDATE_TODO;
  payload: TTodo;
}

interface DeleteTodo {
  type: Types.DELETE_TODO;
  payload: string;
}

export type Action = GetTodos | CreateTodo | UpdateTodo | DeleteTodo;

export const getTodosAction = (todos: TTodo[]): GetTodos => ({
  type: Types.GET_TODOS,
  payload: todos,
});

export const createTodoAction = (todo: TTodo): CreateTodo => ({
  type: Types.CREATE_TODO,
  payload: todo,
});

export const updateTodoAction = (todo: TTodo): UpdateTodo => ({
  type: Types.UPDATE_TODO,
  payload: todo,
});

export const deleteTodoAction = (id: string): DeleteTodo => ({
  type: Types.DELETE_TODO,
  payload: id,
});
