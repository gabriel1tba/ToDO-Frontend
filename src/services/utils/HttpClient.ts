import axios from 'axios';

const HttpClient = axios.create({
  baseURL: `${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}`,
});

export default HttpClient;
