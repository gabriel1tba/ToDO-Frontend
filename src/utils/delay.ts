const delay = (ms = 1000): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));

export default delay;
