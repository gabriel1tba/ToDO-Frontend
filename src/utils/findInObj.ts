// eslint-disable-next-line @typescript-eslint/no-explicit-any
const findInObj = (object: any, value: any): any => {
  // @OBSERVAÇÃO: Remove chaves com valores nulos
  // Object.keys(object).forEach(
  //   (key) => object[key] === null && delete object[key],
  // );

  return Object.values(object).some((v) =>
    typeof v === 'object' && !Array.isArray(object)
      ? findInObj(v, value)
      : typeof v === 'string'
      ? v.toLowerCase().includes(value.toLowerCase())
      : typeof v === 'number'
      ? String(v).includes(value) || (isNaN(v) && isNaN(value))
      : v === value,
  );
};

export default findInObj;
