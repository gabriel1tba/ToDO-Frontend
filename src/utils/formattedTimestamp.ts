export const formattedTimestamp = (timeStamp: string) => {
  return new Date(timeStamp).toLocaleString('pt-BR');
};

export default formattedTimestamp;
