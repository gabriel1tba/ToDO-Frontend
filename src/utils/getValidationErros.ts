import { ValidationError } from 'yup';

type Errors = {
  [key: string]: string;
};

export default function getValidationErros(error: ValidationError): Errors {
  const validationErrors: Errors = {};

  error.inner.forEach((err) => {
    if (typeof err.path === 'string') validationErrors[err.path] = err.message;
  });

  return validationErrors;
}
