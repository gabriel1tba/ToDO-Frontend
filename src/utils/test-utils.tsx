import { render, RenderOptions } from '@testing-library/react';
import { TodoProvider } from 'context/todos';

interface IWrapperProvider {
  children: React.ReactNode;
}

const WrapperProvider = ({ children }: IWrapperProvider) => (
  <TodoProvider>{children}</TodoProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(<WrapperProvider>{ui}</WrapperProvider>, { ...options });

export * from '@testing-library/react';
export { customRender as render };
