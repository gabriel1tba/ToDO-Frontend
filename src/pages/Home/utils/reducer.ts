import { TTodo } from 'services/TodoService/interfaces';

import { Types, Action } from './actions';

const todosReducer = (state: TTodo[], action: Action): TTodo[] => {
  switch (action.type) {
    case Types.GET_TODOS: {
      return action.payload;
    }

    case Types.CREATE_TODO: {
      return [...state, action.payload];
    }

    case Types.UPDATE_TODO: {
      return state.map((todo) => {
        if (action.payload.id === todo.id) {
          todo = { ...action.payload, created_at: todo.created_at };
        }
        return todo;
      });
    }

    case Types.DELETE_TODO: {
      return state.filter((todo) => {
        return todo.id !== action.payload;
      });
    }

    default:
      return state;
  }
};

export default todosReducer;
