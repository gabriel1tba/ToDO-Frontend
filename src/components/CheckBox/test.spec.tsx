import { render, screen } from 'utils/test-utils';

import Checkbox from '.';

describe('<Checkbox />', () => {
  it('should render the Checkbox square border correctly', () => {
    const { container } = render(<Checkbox />);

    expect(screen.getByRole('checkbox')).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).toHaveStyle({
      borderRadius: 0,
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the Checkbox with rounded border correctly', () => {
    render(<Checkbox border="round" />);

    expect(screen.getByRole('checkbox')).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).toHaveStyle({
      borderRadius: '50%',
    });
  });
});
