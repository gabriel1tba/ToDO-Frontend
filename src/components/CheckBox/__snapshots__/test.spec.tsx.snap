// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render the Checkbox square border correctly 1`] = `
.c0 {
  width: 20px;
  height: 20px;
  border-radius: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: solid 2px #2979FF;
}

.c0:checked {
  background-color: #2979FF;
  padding-top: 1px;
  padding-left: 4px;
}

.c0:checked:after {
  content: '';
  display: block;
  height: 9px;
  width: 4.5px;
  border: solid #FFFFFF;
  border-width: 0 2px 2px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<input
  class="c0"
  type="checkbox"
/>
`;
